specification/purview/data-plane/Azure.Analytics.Purview.PDS/preview/2021-01-01-preview/pds.json

```
@@ -0,0 +1,899 @@
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "path",
+            "name": "resourceType",
+            "description": "The resource type.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "path",
+            "name": "resourceName",
+            "description": "The resource name.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/PolicyEventsResponse"
+            }
+          },
+          "304": {
+            "description": "Not Modified"
+          },
+          "400": {
+            "description": "Bad Request",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponseModel"
+            }
+          },
+          "404": {
+            "description": "Not Found",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponseModel"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "PolicyEvents3_Get": {
+            "$ref": "./examples/PolicyEvents3_Get.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "AccountModelBase": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string"
+        },
+        "name": {
+          "type": "string"
+        },
+        "location": {
+          "type": "string"
+        },
+        "tenantId": {
+          "type": "string"
+        },
+        "gatewayEndpoint": {
+          "type": "string"
+        }
+      }
+    },
+    "Consent": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "boolean"
+        },
+        "properties": {
+          "$ref": "#/definitions/Properties"
+        },
+        "excludedScopes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "DatasourcePreferencesResponse": {
+      "type": "object",
+      "properties": {
+        "consent": {
+          "$ref": "#/definitions/Consent"
+        },
+        "accountInfo": {
+          "$ref": "#/definitions/AccountModelBase"
+        }
+      }
+    },
+    "ErrorModel": {
+      "type": "object",
+      "properties": {
+        "code": {
+          "type": "string"
+        },
+        "message": {
+          "type": "string"
+        },
+        "target": {
+          "type": "string"
+        },
+        "details": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ErrorModel"
+          }
+        }
+      }
+    },
+    "ErrorResponseModel": {
+      "type": "object",
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/ErrorModel"
+        }
+      }
```

The error property should be required
```suggestion
      },
      "required": [ "error" ]
```

---

Completed this change.